sc3-prepare.R

```diff
@@ -2,75 +2,114 @@
+  ),
+  make_option(
+    c("-drmax", "--d_region_max"),
+    action = "store",
+    default = NA,
+    type = 'numeric',
+    help = "Defines the maximum number of eigenvectors used for kmeans clustering as a fraction of the total number of cells. Default is 0.07. See SC3 paper for more details."
+  ),
+  make_option(
+    c("-svm_num", "--svm_num_cells"),
+    action = "store",
+    default = NA,
+    type = 'numeric',
+    help = "Number of randomly selected training cells to be used for SVM prediction. Default is NULL."
+  ),
+  make_option(
+    c("-svm_i", "--svm_train_indeces"),
+    action = "store",
+    default = NA,
+    type = 'numeric',
+    help = "A numeric vector defining indeces of training cells that should be used for SVM training. The default is NULL."
+  ),
+  make_option(
+    c("-svm_max", "--svm_max_cells"),
     action = "store",
-    default = 'counts,
-    type = 'characters',
-    help = "A matrix of expression data with rows corresponding to genes and columns corresponding to cells."
+    default = NA,
+    type = 'numeric',
+    help = "Define the maximum number of cells below which SVM are not run."
   ),
   make_option(
-    c("-a", "--annotation"),
+    c("-t", "--num_threads"),
     action = "store",
     default = NA,
+    type = 'numeric',
+    help = "Number of threads/cores to be used in the user's machine."
+  ),
+  make_option(
+    c("-seed", "--rand_seed"),
+    action = "store",
+    default = 8,
     type = 'character',
-    help = "An annotation matrix where rows are the cell names (column names in the expression matrix)"
+    help = "RDS output object"
   ),
   make_option(
     c("-o", "--output"),
     action = "store",
     default = NA,
     type = 'character',
     help = "RDS output object"
-  ),
+  )
 )
 
-opt <- wsc_parse_args(option_list, mandatory = c('cts', 'ann', 'out'))
+
+opt <- wsc_parse_args(option_list)
 
 # Check parameter values
 
-if ( ! dir.exists(opt$cts)){
-  stop((paste('The matrix', opt$cts, 'does not exist')))
+if ( ! dir.exists(opt$SCE)){
+  stop((paste('The SCE object', opt$SCE, 'does not exist')))
 }
 
-# Load SC3 and dependencies
-
-packages <- c("tidyverse", "SC3", "SingleCellExperiment", "scater")
-suppressPackageStartupMessages(lapply(packages, library, character.only = TRUE))
-
 # Create SCE object from data
 
-sce <- SingleCellExperiment(
-    assays = list(
-        counts = as.matrix(opt$cts),
-        logcounts = log2(as.matrix(opt$cts) + 1)
-    ), 
-    colData = opt$ann
-)
-
-# Add gene/transcript symbol
-
-rowData(sce)$feature_symbol <- rownames(sce)
-
-# Remove duplicated entries
-
-sce <- sce[!duplicated(rowData(sce)$feature_symbol), ]
-
-# Define Spike-in's 
-
-isSpike(sce, "ERCC") <- grepl("ERCC", rowData(sce)$feature_symbol)
+sc3_obj <- sc3_prepare(opt$SCE, gene_filter = opt$gene_filter, opt$pct_dropout_min = 10, opt$pct_dropout_max = 90, opt$d_region_min = 0.04,
```

Comment:
Just for consistency with Scater package e.g. https://github.com/ebi-gene-expression-group/bioconductor-scater-scripts/blob/master/scater-normalize.R, could you use SingleCellExperiment for the variable name? It's just going to make it easier to maintain.